server/src/controllers/asset.controller.ts

```diff
@@ -22,6 +23,17 @@ import { UUIDParamDto } from 'src/validation';
 @Controller(Route.ASSET)
 export class AssetController {
   constructor(private service: AssetService) {}
+  @Get('folder/unique-paths')
+  @Authenticated()
+  getUniqueOriginalPaths(@Auth() auth: AuthDto): Promise<string[]> {
+    return this.service.getUniqueOriginalPaths(auth);
+  }
+
+  @Get('folder')
+  @Authenticated()
+  getAssetsByOriginalPath(@Auth() auth: AuthDto, @Query('path') path: string): Promise<AssetResponseDto[]> {
+    return this.service.getAssetsByOriginalPath(auth, path);
+  }
```

Comment:
These endpoints should either be in their own controller, or in the search controller. If we are ever going to allow file/folder level actions, it would make sense to have a dedicated controller IMO. Something like `FolderViewController` at `/views/folder/...`